lib/Optimized64/KeccakP-1600-opt64.c

```diff
@@ -999,28 +999,31 @@ void KeccakP1600_opt64_ExtractBytes(const void *state, unsigned char *data, unsi
     } \
 
 #include <assert.h>
+#include <stdio.h>
 
 size_t KeccakP1600_opt64_12rounds_FastLoop_Absorb(void *state, unsigned int laneCount, const unsigned char *data, size_t dataByteLen)
 {
     size_t originalDataByteLen = dataByteLen;
+
     declareABCDE
     #ifndef KeccakP1600_opt64_fullUnrolling
     unsigned int i;
     #endif
     uint64_t *stateAsLanes = (uint64_t*)state;
     uint64_t *inDataAsLanes = (uint64_t*)data;
 
-    assert(laneCount == 21);
+    // note: for KT256, laneCount will be 17
+    // assert(laneCount == 21);
 
-    #define laneCount 21
+    // #define laneCount 21
```

Comment:
I would add a branch to support both laneCount=21 and laneCount=17.

Reply:
thank you for the suggestion, I have pushed some changes based on [this code from XKCP](https://github.com/XKCP/XKCP/blob/d5f0117f52d1a97223d511347f0053818bc355cf/lib/low/KeccakP-1600/AVX512/C/KeccakP-1600-AVX512.c#L614-L621), and the tests now seem to be working :)